src/marin/processing/classification/classifier.py

```diff
@@ -323,13 +339,238 @@ def __call__(self, batch):
+        self.post_process_fn = post_process_fn
+        self.generated_text_column_name = generated_text_column_name
+        self.prompt_column = prompt_column
+
+        if post_process_fn is None:
+            self.save_original_generation = True
+
+    def ping(self):
+        return self.is_ready
+
+    def _default_post_process_fn(self, generation: str) -> dict[str, Any]:
+        """Default score extractor that looks for numbers 0-5."""
+        import re
+
+        # Try to extract a number from the generation
+        numbers = re.findall(r"\b([0-5])\b", generation.strip())
+        if numbers:
+            return {"score": int(numbers[0])}
+        else:
+            # Default to middle score if no valid number found
+            return {"score": -1}
+
+    def _default_parse_strategy_fn(self, generation: str) -> dict[str, Any]:
+        import re
+
+        # Extract strategy names from lines that start with '##'
+        # Example input:
+        # ## X
+        # ## Y
+        # ## Z
+        matches = list(re.finditer(r"(?m)^\s*##\s*(.+?)\s*$", generation))
+        strategies: list[str] = []
+        seen: set[str] = set()
+
+        for idx, match in enumerate(matches):
+            start = match.end()
+            end = matches[idx + 1].start() if idx + 1 < len(matches) else len(generation)
+            body = generation[start:end].strip()
+            strategy = match.group(1).strip()
+            if body:
+                strategy = f"{strategy}\n{body}"
+            if strategy and strategy not in seen:
+                strategies.append(strategy)
+                seen.add(strategy)
+
+        return {"strategies": strategies}
+
+    def __call__(self, batch: dict[str, Any]) -> dict[str, Any]:
+        """Process a batch of documents."""
+        # Use the text generation pipeline to generate responses
+        batch_with_generations = self.text_generator(batch)
+
+        # Process the generated text to extract classification scores
+        attributes = []
+        generated_texts = batch_with_generations[self.generated_text_column_name]
+
+        if self.post_process_fn is not None:
+            if self.post_process_fn == "score":
+                post_process_fn = self._default_post_process_fn
+            elif self.post_process_fn == "active_reading":
+                post_process_fn = self._default_parse_strategy_fn
+            else:
+                raise ValueError(f"Invalid post_process_fn: {self.post_process_fn}")
+
+            for generated_text in generated_texts:
+                attributes.append({self.attribute_name: post_process_fn(generated_text)})
+
+            batch.update({"attributes": attributes})
+
+        if self.save_original_generation:
+            batch[self.generated_text_column_name] = generated_texts
+
+        # print("Save original generation: ", self.save_original_generation)
+        # print(batch)
+
+        return batch
+
+
+class vLLMClassifierWithDocsAndAttributes(vLLMClassifier):
```

Comment:
This feels kind of weird, like we're over-using inheritance here?  This at least deserves a comment explaining how it's different from the vLLMClassifier.

I'm not sure what "DocsAndAttributes" means here.